greenpithumb/pump_scheduler.py

```diff
@@ -1,27 +1,31 @@
+import datetime
+
+
 class PumpScheduler(object):
     """A wrapper for managing pump scheduling."""
 
-    def __init__(self, clock, sleep_windows):
+    def __init__(self, local_clock, sleep_windows):
         """Creates new PumpScheduler wrapper.
 
         Args:
-            clock: A clock interface
-            sleep_windows: A tuple of tuples, each consisting of a sleep
-                window. Time zone should match that of the clock interface.
+            local_clock: A local clock interface
+            sleep_windows: A list of 2-tuples, each representing a sleep window.
+                Tuple items are datetime.time objects.
         """
-        self._clock = clock
+        self._local_clock = local_clock
         self._sleep_windows = sleep_windows
 
     def is_running_pump_allowed(self):
         """Returns True if OK to run pump, otherwise False."""
```

Comment:
We should define explicitly what makes it OK to run the pump. In particular, we should say that it's not allowed from sleep time (inclusive) until wake time (exclusive).